components/admin_console/billing/delete_workspace/delete_workspace_modal.tsx

```diff
@@ -0,0 +1,354 @@
+
+type Props = {
+    callerCTA: string;
+}
+
+export default function DeleteWorkspaceModal(props: Props) {
+    const dispatch = useDispatch();
+    const openDowngradeModal = useOpenDowngradeModal();
+
+    // License/product checks.
+    const subscription = useGetSubscription();
+    const product = useSelector(getSubscriptionProduct);
+    const isStarter = product?.sku === CloudProducts.STARTER;
+    const isEnterprise = product?.sku === CloudProducts.ENTERPRISE;
+    const license = useSelector(getLicense);
+    const isNotCloud = !isCloudLicense(license);
+
+    // Starter product for downgrade purposes.
+    const starterProduct = useSelector((state: GlobalState) => {
+        return Object.values(state.entities.cloud.products || {}).find((product) => {
+            return product.sku === CloudProducts.STARTER;
+        });
+    });
+
+    // Get usage information in an attempt to defer customer from deleting.
+    const usage = useGetUsage();
+    const totalFileSize = fileSizeToString(usage.files.totalStorage);
+    const totalMessages = useSelector((state: GlobalState) => {
+        if (!state.entities.admin.analytics) {
+            return 0;
+        }
+        return state.entities.admin.analytics[StatTypes.TOTAL_POSTS];
+    });
+
+    // Creates a success modal for deletions, with button to redirect to mattermost.com.
+    const deleteSuccessModal = () => {
+        const handleButtonClick = () => {
+            window.open(MattermostLink, '_blank');
+        };
+
+        const title = (
+            <FormattedMessage
+                defaultMessage={'Your workspace has been deleted'}
+                id={'admin.billing.deleteWorkspace.successModal.title'}
+            />
+        );
+
+        const subtitle = (
+            <FormattedMessage
+                id={'admin.billing.deleteWorkspace.successModal.subtitle'}
+                defaultMessage={'Your workspace has now been deleted. Thank you for being a customer.'}
+            />
+        );
+
+        const buttonText = (
+            <FormattedMessage
+                id='delete_success_modal.button_text'
+                defaultMessage={'Go to mattermost.com'}
+            />
+        );
+
+        return (
+            <ResultModal
+                primaryButtonText={buttonText}
+                primaryButtonHandler={handleButtonClick}
+                identifier={ModalIdentifiers.DELETE_WORKSPACE_SUCCESS}
+                subtitle={subtitle}
+                title={title}
+                icon={
+                    <PaymentSuccessStandardSvg
+                        width={444}
+                        height={313}
+                    />
+                }
+                contactSupportButtonVisible={false}
+            />
+        );
+    };
+
+    // Creates a failure modal for deletions, with a contact support (technical contact) and try again buttons.
+    const deleteFailureModal = () => {
```

Comment:
Rather than defining 3 modals and flipping between them (and if I'm reading right, also flipping between the non-fullscreen GenericModal and the FullscreenModal used by ResultModal), can we instead just change the content displayed within this one modal? It might be good to have separate components and component files for the success and fail states, as well. Especially since the action of this failure modal closes the fail modal and goes back to the delete modal, this seems more appropriate.

Reply:
Just for clarity, are you saying you'd prefer the delete failure, success and progress modals should be in their own components or are you looking for one modal that has all the content and can change between failure and success ( and maybe progress too)? I'd imagine that component would take a boolean prop to determine if it's a failure or a success.

Reply:
A single modal, and each "screen" can be its own component in its own file. I'm thinking something along the lines of how the self hosted purchase modal does it for the success/progress/error states, using the component's internal `useState` hooks to track whether its a success or failure or in progress, and then use that to decide which to display within the modal: https://github.com/mattermost/mattermost-webapp/blob/master/components/self_hosted_purchase_modal/index.tsx#L702-L720

Reply:
Gotcha, makes sense! I'll get this going, thanks :)